spec/features/locations/edit_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+describe "location edit page", type: :feature do
+  before do
+    @location = FactoryBot.create :location
+  end
+
+  it "exists at 'edit_author_path' and render without error" do
+    visit edit_location_url(@location)
+    page.fill_in 'location[name]', with: 'Updated_name'
+    page.fill_in 'location[details]', with: 'Updated_details'
+    page.fill_in 'location[location_latitude]', with: 14.0
+    page.fill_in 'location[location_longitude]', with: 20.0
+    find('input[type="submit"]').click
+
+    expect(Location.where(name: "Updated_name", details: "Updated_details", location_latitude: 14.0,
+                          location_longitude: 20.0)).to exist
+  end
+
+  it "can update photo" do
+    @location.location_photo.attach(
+      io: File.open('app/assets/images/default-profile-picture.png'),
+      filename: 'default-profile-picture.png',
+      content_type: 'image/png'
+    )
+    filename = @location.location_photo.filename
+    @location.location_photo.detach
+    visit location_path(@location)
+    expect(page).not_to have_css "img[alt='#{filename}']"
+
+    visit edit_location_path(@location)
+    page.attach_file('location[location_photo]', 'app/assets/images/default-profile-picture.png')
+    page.find('input[type=submit][name=commit]').click
+
+    visit location_path(@location)
+    expect(page).to have_css "img[alt='#{filename}']"
+  end
```

Comment:
weird test, why attach and then detach immediately? Why not just hardcode filename as it is done 3 lines above
Otherwise, not very pretty to test other route (show) in edit_spec, is this the right location? (but in edit_spec fine with me, no better idea; isn't this rather an integration test?)

Reply:
The reason for attaching and detaching is, that the actual filename is generated when attaching (it's a long number and not just default-profile-picture). Therefore we have to attach it first to find out the generated filename.

If you have you have concrete ideas how to improve this test, we are open to them :)
Otherwise we will leave the test as is.